Gems/RecastNavigation/Code/Source/Misc/RecastNavigationDebugDraw.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <DebugDraw.h>
+#include <AzCore/std/containers/vector.h>
+#include <Misc/RecastHelpers.h>
+#include <Misc/RecastSmartPointer.h>
+
+namespace RecastNavigation
+{
+    //! Recast library specific debug draw that captures and draws the various debug overlays.
+    //! See @duDebugDraw for documentation of the methods.
+    class RecastNavigationDebugDraw final : public duDebugDraw
+    {
+    public:
+        explicit RecastNavigationDebugDraw(bool drawLines = false);
+
+        //! duDebugDraw overrides...
+        //! @{
+        //! Not implemented
+        void depthMask([[maybe_unused]] bool state) override {}
+        //! Not implemented
+        void texture([[maybe_unused]] bool state) override {}
```

Comment:
question: Should we assert or error on these in case they are called? Or is it fine to fail silently?

Reply:
It is fine. It's just a debug draw of the nav mesh. 

In fact, so far I have not seen Recast libraries call these at all in my unit test coverage.